src/greynirseq/nicenlp/tasks/document_translation_from_pretrained_bart.py

```diff
@@ -0,0 +1,385 @@
+        # langcode and translation direction, e.g. "en:en_XX,is:is_IS"
+        language_mappings_pairs = cfg.data_language_mappings.split(",")
+        language_mappings = {}
+        for pair in language_mappings_pairs:
+            parts = pair.split(":")
+            if len(parts) != 2:
+                raise ValueError("Invalid language mapping: {}".format(pair))
+            language_mappings[parts[0]] = parts[1]
+
+        return cls(cfg, the_dict, language_mappings)
+
+    def load_dataset(self, split: str, epoch=1, combine=False, **kwargs):
+        """Load a given dataset split.
+
+        This function is called once per train split (e.g., "train") and multiple times per validation split.
+        This function has deviated a lot from the original fairseq implementation.
+
+        Args:
+            split (str): The value of --train-subset OR
+                --valid-subset which has been split on "," OR
+                --test-subset which has been split on "," CLI args.
+                Each is a comma-separated list of dataset names.
+                This method is called separately for each subset.
+        """
+        self.cfg = cast(DocumentTranslationFromPretrainedBARTConfig, self.cfg)
+        # this is for sharding
+        paths = utils.split_paths(self.cfg.data)  # type: ignore
+        assert len(paths) > 0
+        data_dir_path = paths[(epoch - 1) % len(paths)]
+
+        # if a split contains a comma, we should crash - since that is no longer supported
+        assert "," not in split, "Split should not contain a comma"
+        # now lets list all the 'jsonl' files in the data dir
+        jsonl_files = glob.glob(os.path.join(data_dir_path, "*.jsonl"))
+        jsonl_files_paths = [pathlib.Path(path) for path in jsonl_files]
+        # now we gather all the files which start with the same string as the split
+        split_files = [path for path in jsonl_files_paths if path.stem.startswith(split)]
+        for path in split_files:
+            logger.info(f"Found file {path}")
+
+        def metadata_from_filename(path: pathlib.Path):
+            """Extract translation metadata from filename.
+
+            There are 3 types of files:
+            1. {name}.{lang1}-{lang2}.{lang1}.jsonl
+            2. {name}.{lang1}-{lang2}.{lang2}.jsonl
+            3. {name}.{lang1}-{lang2}.align.jsonl [optional]
+            """
+            name, direction, file_type = path.stem.split(".")  # .stem removes the jsonl extension
+            lang1, lang2 = direction.split("-")
+            assert lang1 in self.language_mappings
+            assert lang2 in self.language_mappings
+            assert file_type in [lang1, lang2, "align"]
+            if file_type == "align":
+                file_type = "align"
+            elif file_type == lang1:
+                file_type = "src"
+            elif file_type == lang2:
+                file_type = "tgt"
+            return {
```

Comment:
This would be better as a namedtuple (even if the namedtuple constructor is defined locally).